src/core/raster/qgsrasterattributetable.cpp

```diff
@@ -296,24 +317,91 @@ bool QgsRasterAttributeTable::insertField( const Field &field, int position, QSt
     it->insert( realPos, defaultValue );
   }
 
-  setIsDirty( true );
+  // Set/change the table type from the value field type
+  if ( field.usage == Qgis::RasterAttributeTableFieldUsage::MinMax )
```

Comment:
use a switch here?

Reply:
I would say no, this is a particular check on those particular value field roles only. To me it is much clearer than a switch where we are supposed to handle multiple roles (I believe I have used switches on all other cases).